File: typescript/src/agent/reasoning.ts
// =============================================================================
// Agent Layer — Reasoning Helpers
// =============================================================================

import type { OROutputItem } from "./types.js";

/**
 * Extract reasoning summary text from response output items.
 * Returns the concatenated summary text, or undefined if no reasoning present.
 */
export function extractReasoning(output: OROutputItem[]): string | undefined {
  for (const item of output) {
    if (item.type === "reasoning" && item.summary.length > 0) {
      const text = item.summary.map((s) => s.text).join("");
      return text || undefined;
    }
  }
  return undefined;
}

/**
 * Push a reasoning string into the accumulator array.
 */
export function accumulateReasoning(collected: string[], text: string): void {
  collected.push(text);
}

File: python/src/opperai/agent/_reasoning.py
"""Agent SDK — Reasoning extraction helpers."""

from __future__ import annotations

from typing import Any


def extract_reasoning(output_items: list[dict[str, Any]]) -> str | None:
    """Extract reasoning text from response output items."""
    for item in output_items:
        if item.get("type") == "reasoning" and item.get("summary"):
            parts = item["summary"]
            text = "".join(part.get("text", "") for part in parts)
            if text:
                return text
    return None


def accumulate_reasoning(collected: list[str], text: str) -> None:
    """Append reasoning text to the collected list."""
    collected.append(text)
